Library/Core/UnoCore/Backends/CPlusPlus/Uno/Support.h

```diff
@@ -42,8 +40,6 @@ uBase::Vector2i uInt2ToXliVector2i(const @{int2}& vec);
 
 @{float2} uFloat2FromXliVector2(const uBase::Vector2& vec);
 uBase::Vector2 uFloat2ToXliVector2(const @{float2}& vec);
-
-@{Uno.Buffer} uBufferFromXliDataAccessor(const uBase::DataAccessor* data);
```

Comment:
I think we should keep these things around until `Uno.Buffer` has actually been removed, otherwise we break existing code...

Reply:
The only user of this method was removed in https://github.com/fuse-open/uno/pull/80/commits/a5d165f2a939fb90c2650b11fdb2bb1f107bae30#diff-4aed3dbdda54ba506cf8d9e73348defcL91, prior to the commit that removes the method itself.

The method is also `uno-base` specific, not used by `fuselibs`, `Fuse.Entities` or `fuse-studio`, and I think it's very unlikely that any third-parties are using this.

Also, since `Support.h` must be included separately, it is not really part of the "core C++ API" I would say. I don't think it will be a problem to remove this.

Only `U_BUFFER_PTR` and `U_BUFFER_SIZE` were used by `Experimental.TextureLoader` in `fuselibs`, so those are kept for now. When https://github.com/mortend/fuselibs/tree/replace-buffer is merged, these macros won't be used anymore either...

Reply:
I wouldn't be so sure. There's a lot of foreign code floating around...

Reply:
OK, just let me fill in a bit why I worry about things like this. Experience from our release-management in the past has shown that some people tend to use code we have no idea about, and we don't find out about it until when it's way too late. The end result can be that the product looks unreliable, and that old, not actively maintained code tends to not work. Which is a problem for a product that has a lot of not-really-maintained 3rd-party out there... Just looking through the code that *we* see for compatibility issues has bitten us in the past.

You might be right in this particular case, it's just that the argument that this is unused because it's not used in fuselibs, Fuse.Entites and fuse-studio, well, that's been proven to not be enough before.

Now, I'm not proposing that we stop changing or removing APIs entirely, just that things like this should go through a normal deprecation cycle, and be documented as such in the changelog so people have a chance to catch up.

Reply:
That's a thorough write-up, thank you! I understand your concerns, and in many cases I agree.

I'm sometimes a bit trigger happy when it comes to deleting old (and badly designed) code - the least risky route is of course to not delete the code, and also the least satisfying.

Anyway, here comes a rebase with the deprecations.